spec/task_router/workflow_builder_spec.rb

```diff
@@ -0,0 +1,387 @@
+      expect(defaultTarget.queue).to eq(everyoneQueue)
+      expect(defaultTarget.priority).to eq(priority)
+      expect(defaultTarget.timeout).to eq(timeout)
+      expect(defaultTarget.expression).to eq(filterWorkerExpression)
+    end
+
+    it 'should construct simple rule' do
+      salesQueue = "WQf6724bd5005b30eeb6ea990c3e59e536"
+      salesTarget = Twilio::TaskRouter::WorkflowRuleTarget.new salesQueue
+      expression = "type=='sales'"
+      friendlyName = "Sales"
+      salesTargets = [salesTarget]
+      salesRule = Twilio::TaskRouter::WorkflowRule.new expression, salesTargets, friendlyName
+      expect(salesRule.expression).to eq(expression)
+      expect(salesRule.friendly_name).to eq(friendlyName)
+      expect(salesRule.targets[0]).to eq(salesTarget)
+    end
+
+    it 'should construct a full workflow' do
+      salesQueue = "WQf6724bd5005b30eeb6ea990c3e59e536"
+      marketingQueue = "WQ8c62f84b61ccfa6a333757cd508f0aae"
+      supportQueue = "WQ5940dc0da87eaf6e3321d62041d4403b"
+      everyoneQueue = "WQ6d29383312b24bd898a8df32779fc043"
+
+      defaultTarget = Twilio::TaskRouter::WorkflowRuleTarget.new everyoneQueue
+
+      salesTarget = Twilio::TaskRouter::WorkflowRuleTarget.new salesQueue
+      salesRule = Twilio::TaskRouter::WorkflowRule.new "type=='sales'", [salesTarget], 'Sales'
+
+      marketingTarget = Twilio::TaskRouter::WorkflowRuleTarget.new marketingQueue
+      marketingRule = Twilio::TaskRouter::WorkflowRule.new "type=='marketing'", [marketingTarget], 'Marketing'
+
+      supportTarget = Twilio::TaskRouter::WorkflowRuleTarget.new supportQueue
+      supportRule = Twilio::TaskRouter::WorkflowRule.new "type=='support'", [supportTarget], 'Support'
+
+      rules = [salesRule, marketingRule, supportRule]
+
+      config = Twilio::TaskRouter::WorkflowConfiguration.new rules, defaultTarget
+      json = config.to_json
+
+      expected = "{
+				\"task_routing\":{
+				  \"filters\":[
+					 {
+						\"expression\":\"type=='sales'\",
+						\"targets\":[
+						   {
+							  \"queue\":\"WQf6724bd5005b30eeb6ea990c3e59e536\"
+						   }
+						],
+						\"friendly_name\":\"Sales\"
+					 },
+					 {
+						\"expression\":\"type=='marketing'\",
+						\"targets\":[
+						   {
+							  \"queue\":\"WQ8c62f84b61ccfa6a333757cd508f0aae\"
+						   }
+						],
+						\"friendly_name\":\"Marketing\"
+					 },
+					 {
+						\"expression\":\"type=='support'\",
+						\"targets\":[
+						   {
+							  \"queue\":\"WQ5940dc0da87eaf6e3321d62041d4403b\"
+						   }
+						],
+						\"friendly_name\":\"Support\"
+					 }
+				  ],
+				  \"default_filter\":{
+					 \"queue\":\"WQ6d29383312b24bd898a8df32779fc043\"
+				  }
+				}
+			}"
+
+      # replace all new tabs, new lines and spaces
+      expected = expected.gsub /\t/, ''
+      expected = expected.gsub /\n/, ''
+      expected = expected.gsub /\s+/, ''
```

Comment:
you can just `to_json(from_json(expected))` to get rid of all the tabs and spaces